contracts/Checkpointing.sol

```diff
@@ -15,10 +15,14 @@ library Checkpointing {
     uint256 constant MAX_UINT64 = uint256(uint64(-1));
 
     function add192(History storage self, uint64 time, uint192 value) internal {
-        if (self.history.length == 0 || self.history[self.history.length - 1].time < time) {
+        uint256 length = self.history.length;
+
+        if (length == 0) {
+            self.history.push(Checkpoint(time, value));
+        } else if (self.history[length - 1].time < time) {
```

Comment:
Is this split part of the gas optimization? Solidity has boolean short-circuit evaluation to optimize them, it shouldn't cost less gas than having it as it was before.

Reply:
Yes, I was surprised too first time I saw this, and definitely the code looks uglier, but I tried With Remix and this way it was cheaper. I haven't looked at the EVM code, but with this example, with and without optimizations, and on 3 branches, the result was better was better with the split:
https://gist.github.com/bingen/120c58cdcd6e46e32c08ef61239783ca
But as the gain was really tiny (in some cases only 1 or 3 gas), I can revert it.